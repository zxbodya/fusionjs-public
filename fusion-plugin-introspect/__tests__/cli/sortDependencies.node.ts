// @flow
import {sortDependencies} from '../../src/shared/sortDependencies.js';

test('sortDependencies', async () => {
  const data = sortDependencies([
    {
      name: 'A',
      type: 'both',
      sources: [{type: 'plugin', source: 'foo/bar:1:2'}],
      dependencies: ['B'],
    },
    {
      name: 'B',
      type: 'both',
      sources: [{type: 'plugin', source: 'foo/bar:3:4'}],
      dependencies: ['C'],
    },
    {
      name: 'C',
      type: 'both',
      sources: [{type: 'plugin', source: 'foo/bar:5:6'}],
      dependencies: [],
    },
  ]);
  expect(data).toMatchSnapshot();
});
