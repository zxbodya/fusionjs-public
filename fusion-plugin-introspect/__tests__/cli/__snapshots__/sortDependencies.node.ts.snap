// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortDependencies 1`] = `
Array [
  Object {
    "dependencies": Array [],
    "name": "C",
    "sources": Array [
      Object {
        "source": "foo/bar:5:6",
        "type": "plugin",
      },
    ],
    "type": "both",
  },
  Object {
    "dependencies": Array [
      "C",
    ],
    "name": "B",
    "sources": Array [
      Object {
        "source": "foo/bar:3:4",
        "type": "plugin",
      },
    ],
    "type": "both",
  },
  Object {
    "dependencies": Array [
      "B",
    ],
    "name": "A",
    "sources": Array [
      Object {
        "source": "foo/bar:1:2",
        "type": "plugin",
      },
    ],
    "type": "both",
  },
]
`;
